pkg/engine/handlers/validation/validate_cel.go

```diff
@@ -76,20 +81,21 @@ func (h validateCELHandler) Process(
 	validations := rule.Validation.CEL.Expressions
 	auditAnnotations := rule.Validation.CEL.AuditAnnotations
 
-	optionalVars := cel.OptionalVariableDeclarations{HasParams: hasParam, HasAuthorizer: false}
+	optionalVars := cel.OptionalVariableDeclarations{HasParams: hasParam, HasAuthorizer: true}
+	expressionOptionalVars := cel.OptionalVariableDeclarations{HasParams: hasParam, HasAuthorizer: false}
 	// compile CEL expressions
 	compiler, err := celutils.NewCompiler(validations, auditAnnotations, matchConditions, variables)
 	if err != nil {
 		return resource, handlers.WithError(rule, engineapi.Validation, "Error while creating composited compiler", err)
 	}
 	compiler.CompileVariables(optionalVars)
 	filter := compiler.CompileValidateExpressions(optionalVars)
-	messageExpressionfilter := compiler.CompileMessageExpressions(optionalVars)
+	messageExpressionfilter := compiler.CompileMessageExpressions(expressionOptionalVars)
 	auditAnnotationFilter := compiler.CompileAuditAnnotationsExpressions(optionalVars)
 	matchConditionFilter := compiler.CompileMatchExpressions(optionalVars)
 
 	// newMatcher will be used to check if the incoming resource matches the CEL preconditions
-	newMatcher := matchconditions.NewMatcher(matchConditionFilter, nil, "", "", "")
+	newMatcher := matchconditions.NewMatcher(matchConditionFilter, nil, policyKind, "", policyName)
```

Comment:
Does it make sense to check policy namespace though currently we only support cel in a ClusterPolicy? Do we block namespace policy creation if cel subrule is defined?

Reply:
Namespaced policies can be created with cel subrules but we can't generate validating admission policies from it. In that case Kyverno engine handles the resource validation itself.